sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ContentDownloader.java

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.communication.callingserver;
+
+import com.azure.communication.callingserver.implementation.Constants;
+import com.azure.communication.callingserver.models.CallingServerResponseException;
+import com.azure.communication.callingserver.models.ParallelDownloadOptions;
+import com.azure.communication.callingserver.models.ProgressReporter;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRange;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.SignalType;
+import reactor.core.scheduler.Schedulers;
+import reactor.util.function.Tuple2;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Function;
+
+import static java.lang.StrictMath.toIntExact;
+
+class ContentDownloader {
+    private final String resourceEndpoint;
+    private final HttpPipeline httpPipeline;
+    private final ClientLogger logger;
+
+    ContentDownloader(String resourceEndpoint, HttpPipeline httpPipeline,
+                             ClientLogger logger) {
+        this.resourceEndpoint = resourceEndpoint;
+        this.httpPipeline = httpPipeline;
+        this.logger = logger;
```

Comment:
I wanted to use the calling class's logger instead of this auxiliary's one. 
In that way, the error may be in AsyncClient or Client, and not always the same ContentDownloader.
Please advise if this is not a good practice.